src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -387,6 +392,62 @@ public FeatureDataSource( final File featureFile, final FeatureCodec<T, ?> codec
     }
 
     /**
+     * Returns the sequence dictionary created from the index or null if that fails.
+     */
+    public SAMSequenceDictionary getSequenceDictionary() {
+        SAMSequenceDictionary dict = null;
+        final Object header = getHeader();
+        if (header instanceof VCFHeader) {
+            dict = ((VCFHeader) header).getSequenceDictionary();
+        }
+        return dict == null || dict.isEmpty() ? createSequenceDictionaryFromContigList() : dict;
+    }
+
+    /**
+     * Load the index from disk, checking for out of date indexes and old versions
+     * @return an Index, or null if we're unable to load
+     */
+    private Index loadIndexFromDisk() {
```

Comment:
This method should not be embedded within `FeatureDataSource` like this. Extract to a public method that takes a `featureFile` as an argument in an appropriate utils class (or create `IndexUtils` if we don't already have one), and add a direct unit test for it.